pkg/store/promproxy.go

```diff
@@ -134,10 +131,9 @@ func (p *PrometheusProxy) Series(ctx context.Context, r *storepb.SeriesRequest)
 	res := &storepb.SeriesResponse{
 		Series: make([]storepb.Series, 0, len(m.Data.Result)),
 	}
-	ext := p.externalLabels()
 
 	for _, e := range m.Data.Result {
-		lset := translateAndExtendLabels(e.Metric, ext)
+		lset := translateAndExtendLabels(e.Metric, p.externalLabels())
```

Comment:
I think we don't want to call here on every iteration. Otherwise we have a semantic race and could be attaching different label sets on parts of the result if the external labels get updated.

Reply:
ups, this shouldn't be in a loop, you're right